src/components/application_manager/test/commands/mobile/change_registration_test.cc

```diff
@@ -419,18 +456,65 @@ TEST_F(ChangeRegistrationRequestTest, Run_IsLanguageSupportedByUITrue_SUCCESS) {
   command_->Run();
 }
 
+TEST_F(ChangeRegistrationRequestTest,
+       Run_IsLanguageSupportedByVRLanguagesEmpty_UNSUCCESS) {
+  CreateDefaultMessage(message_);
+  const utils::custom_string::CustomString name(kAppName2);
+  application_set_.insert(app_);
+
+  EXPECT_CALL(mock_app_manager_, application(kConnectionKey))
+      .WillOnce(Return(app_));
+  EXPECT_CALL(mock_app_manager_, applications())
+      .WillOnce(Return(applications_));
+  EXPECT_CALL(*app_, name()).WillOnce(ReturnRef(name));
+
+  MessageSharedPtr vr_synonyms_message(CreateMessage());
+  (*vr_synonyms_message)[am::strings::msg_params][am::strings::vr_synonyms][0] =
+      kVrSynonyms2;
+  smart_objects::SmartObject& vr_synonyms =
+      (*vr_synonyms_message)[am::strings::msg_params][am::strings::vr_synonyms];
+  EXPECT_CALL(*app_, vr_synonyms())
+      .WillOnce(Return(static_cast<smart_objects::SmartObject*>(&vr_synonyms)));
+  EXPECT_CALL(mock_app_manager_, hmi_capabilities())
+      .WillRepeatedly(ReturnRef(mock_hmi_capabilities_));
+
+  (*vr_synonyms_message)[am::strings::msg_params]
+                        [am::strings::hmi_display_language][0] = 1;
+  (*vr_synonyms_message)[am::strings::msg_params]
+                        [am::strings::hmi_display_language][1] = 2;
+  smart_objects::SmartObject* ui_languages =
+      &((*vr_synonyms_message)[am::strings::msg_params]
+                              [am::strings::hmi_display_language]);
+  EXPECT_CALL(mock_hmi_capabilities_, ui_supported_languages())
+      .WillOnce(Return(ui_languages));
+
+  smart_objects::SmartObjectSPtr vr_languages = CreateMessage();
+  (*vr_languages) = 0;
```

Comment:
@okozlovlux do we have better syntax for creating smart object with 0? This one is ugly.